deepchem/models/__init__.py

```diff
@@ -148,8 +154,195 @@ def predict(self, dataset, transformers=[]):
+    # Special case to handle singletasks.
+    if n_tasks == 1:
+      y_pred = np.reshape(y_pred, (n_samples,)) 
     return y_pred
 
+  def predict_grad(self, dataset, transformers=[]):
+    """
+    Uses self to calculate gradient on provided Dataset object.
+
+    TODO(rbharath): Should we assume each model has meaningful gradients to
+    predict? Should this be a subclass for PhysicalModel or the like?
+
+    Returns:
+      y_pred: numpy ndarray of shape (n_samples,)
+    """
+    grads = []
+    batch_size = self.model_params["batch_size"]
+    for (X_batch, y_batch, w_batch, ids_batch) in dataset.iterbatches(batch_size):
+      energy_batch = self.predict_on_batch(X_batch)
+      grad_batch = self.predict_grad_on_batch(X_batch)
+      grad_batch = undo_grad_transforms(grad_batch, energy_batch, transformers)
+      grads.append(grad_batch)
+    grad = np.vstack(grads)
+  
+    return grad
+
+  def evaluate_error(self, dataset, transformers=[]):
+    """
+    Evaluate the error in energy and gradient components, forcebalance-style.
+
+    TODO(rbharath): This looks like it should be a subclass method for a
+    PhysicalMethod class. forcebalance style errors aren't meaningful for most
+    chem-informatic datasets.
+    """
+    y_preds = []
+    y_train = []
+    batch_size = self.model_params["batch_size"]
+    for (X_batch, y_batch, w_batch, ids_batch) in dataset.iterbatches(batch_size):
+
+      y_pred_batch = self.predict_on_batch(X_batch)
+      y_pred_batch = np.reshape(y_pred_batch, y_batch.shape)
+
+      y_pred_batch = undo_transforms(y_pred_batch, transformers)
+      y_preds.append(y_pred_batch)
+
+      y_batch = undo_transforms(y_batch, transformers)
+      y_train.append(y_batch)
+
+    y_pred = np.vstack(y_preds)
+    y = np.vstack(y_train)
+
+    n_samples, n_tasks = len(dataset), len(self.tasks)
+    n_atoms = int((n_tasks-1)/3)
+
+    y_pred = np.reshape(y_pred, (n_samples, n_tasks)) 
+    y = np.reshape(y, (n_samples, n_tasks))
+    grad = y_pred[:,1:]
+    grad_train = y[:,1:]
+
+    energy_error = y[:,0]-y_pred[:,0]
+    # convert Hartree to kJ/mol
+    energy_error = np.sqrt(np.mean(energy_error*energy_error))*2625.5002
+ 
+    grad = np.reshape(grad, (n_samples, n_atoms, 3))
+    grad_train = np.reshape(grad_train, (n_samples, n_atoms, 3))    
+  
+    grad_error = grad-grad_train
+    # convert Hartree/bohr to kJ/mol/Angstrom
+    grad_error = np.sqrt(np.mean(grad_error*grad_error))*4961.47596096
+
+    print("Energy error (RMSD): %f kJ/mol" % energy_error)
+    print("Grad error (RMSD): %f kJ/mol/A" % grad_error)
+    
+    return energy_error, grad_error
+
+  def evaluate_error_class2(self, dataset, transformers=[]):
+    """
+    Evaluate the error in energy and gradient components, forcebalance-style.
+
+    TODO(rbharath): Should be a subclass PhysicalModel method. Also, need to
+    find a better name for this method (class2 doesn't tell us anything about the
+    semantics of this method.
```

Comment:
@rbharath I agree. Along the lines of my above comment, this should be the compute error function of the `AtomicRegressor` class that trains only on energy but predicts forces with tf.gradients.